Remotion/Data/DomainObjects.UnitTests/Infrastructure/ObjectPersistence/RootPersistenceStrategyTest.cs

```diff
@@ -15,29 +15,504 @@
+                Assert.That(containerArray[1].ID, Is.EqualTo(objectID2));
+              })
+          .Verifiable();
+
+      _rootPersistenceStrategy.PersistData(data);
+
+      _persistenceServiceMock.Verify();
+    }
+
+    private ILoadedObjectDataProvider SetupLoadedObjectDataProviderStubWithDataContainers (
+        DataContainer returnedDataContainer,
+        DataContainer returnedNotYetLoadedDataContainer,
+        ILoadedObjectData loadedObjectData)
+    {
+      var loadedObjectDataProviderStub = new Mock<ILoadedObjectDataProvider>();
+      loadedObjectDataProviderStub
+          .Setup(_ => _.GetLoadedObject(returnedDataContainer.ID))
+          .Returns(loadedObjectData);
+      loadedObjectDataProviderStub
+          .Setup(_ => _.GetLoadedObject(returnedNotYetLoadedDataContainer.ID))
+          .Returns((ILoadedObjectData)null);
+      return loadedObjectDataProviderStub.Object;
+    }
+
+    private Mock<ILoadedObjectData> SetupLoadedObjectDataStubWithDataContainer (DataContainer returnedDataContainer)
+    {
+      var loadedObjectDataStub = new Mock<ILoadedObjectData>();
+      loadedObjectDataStub.Setup(_ => _.ObjectID).Returns(returnedDataContainer.ID);
+      return loadedObjectDataStub;
+    }
+
+    private IQuery SetupQueryAndInfrastructureForQueryType (QueryType queryType, bool createReadOnlyStorageProvider)
+    {
+      _unitTestStorageProviderStubDefinition = new UnitTestStorageProviderStubDefinition(
+          "storageProviderDefinition",
+          _rdbmsStorageObjectFactoryStub.Object,
+          new[] { typeof(Customer), typeof(Order) });
+
+      var queryStub = new Mock<IQuery>();
+      queryStub.Setup(_ => _.QueryType).Returns(queryType);
+      queryStub.Setup(_ => _.StorageProviderDefinition).Returns(_unitTestStorageProviderStubDefinition);
+
+      _storageSettingsMock
+          .Setup(_ => _.GetStorageProviderDefinition(_unitTestStorageProviderStubDefinition.Name))
+          .Returns(_unitTestStorageProviderStubDefinition);
+
+      return queryStub.Object;
+    }
+
+    private void SetupRdbmsStorageObjectFactoryStubToCreateStorageProvider (bool createReadOnlyStorageProvider)
+    {
+      if (createReadOnlyStorageProvider)
+      {
+        _rdbmsStorageObjectFactoryStub
+            .Setup(_ => _.CreateReadOnlyStorageProvider(_unitTestStorageProviderStubDefinition, It.IsAny<IPersistenceExtension>()))
+            .Returns(_storageProviderStub.Object);
+      }
+      else
+      {
+        _rdbmsStorageObjectFactoryStub
+            .Setup(_ => _.CreateStorageProvider(_unitTestStorageProviderStubDefinition, It.IsAny<IPersistenceExtension>()))
+            .Returns(_storageProviderStub.Object);
+      }
+    }
+
+    private IReadOnlyStorageProviderManager ResolveAnyStorageAccessProvider (bool readOnlyStorageAccess)
+    {
+      if (readOnlyStorageAccess)
+        return It.IsAny<ReadOnlyStorageProviderManager>();
+      else
+        return It.IsAny<StorageProviderManager>();
     }
```

Comment:
The method name should be based on `StorageProviderManager`, not `StorageAccessProvider`, because that is what we get as return value.
Instead of "ResolveAny", what about "GetArgumentConstraint" or just "Get"? I understand that we use It.IsAny<T>, but we don't resolve anything, we get an argument constraint. However, `GetArgumentConstraintForStorageProviderManager(readOnlyStorageAccess)` is unwieldy, therefore I suggest `GetStorageProviderManager(readOnlyStorageAccess)`